File: CivitasJava/src/JuegoTexto/Controlador.java
package JuegoTexto;

import Civitas.CivitasJuego;
import Civitas.Diario;
import Civitas.OperacionesJuego;
import Civitas.Respuestas;
import Civitas.SalidasCarcel;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Scanner;
import Civitas.Casilla;
import Civitas.GestionesInmobiliarias;
import Civitas.Jugador;
import Civitas.TituloPropiedad;
import Civitas.OperacionInmobiliaria;

public class Controlador {

    private CivitasJuego juego;
    private VistaTextual vista;

    Controlador(CivitasJuego juego, VistaTextual vista) {
        this.juego = juego;
        this.vista = vista;
    }

    void juega() {
        OperacionesJuego operacion;
        Respuestas resp;
        GestionesInmobiliarias gest;
        int numProp;
        int indGest;
        
        vista.setCivitasJuego(juego);
        
        while(!this.juego.finalDelJuego()){
            vista.actualizarVista();
            vista.pausa();
            operacion = juego.siguientePaso();

            if (operacion != OperacionesJuego.PASAR_TURNO)
                vista.mostrarEventos();
            
            if (!juego.finalDelJuego()) {
                switch (operacion) {
                    case COMPRAR:
                        resp = vista.comprar();
                        if (resp == Respuestas.SI) {
                            juego.comprar();
                        }
                        juego.siguientePasoCompletado(operacion);
                        break;
                    case GESTIONAR:
                        vista.gestionar();
                        indGest = vista.getGestion();
                        numProp = vista.getPropiedad();
                        gest = GestionesInmobiliarias.values()[indGest];
                        OperacionInmobiliaria opInmobiliaria = new OperacionInmobiliaria(gest, numProp);
                        System.out.println(indGest);
                        switch (indGest) {
                            case 1:
                                juego.vender(opInmobiliaria.getNumPropiedad());
                                break;
                            case 2:
                                juego.hipotecar(opInmobiliaria.getNumPropiedad());
                                break;
                            case 3:
                                juego.cancelarHipoteca(opInmobiliaria.getNumPropiedad());
                                break;
                            case 4:
                                juego.construirCasa(opInmobiliaria.getNumPropiedad());
                                break;
                            case 5:
                                juego.construirHotel(opInmobiliaria.getNumPropiedad());
                                break;
                            case 0:
                                juego.siguientePasoCompletado(operacion);
                                break;
                        }
<<<<<<< HEAD
                        juego.siguientePasoCompletado(operacion);
=======
>>>>>>> 1a99332136297a7a0cc0fd85eceba1402b9c5840
                        break;
                    case SALIR_CARCEL:
                        switch (vista.salirCarcel()) {
                            case PAGANDO:
                                juego.salirCarcelPagando();
                                break;
                            case TIRANDO:
                                juego.salirCarcelTirando();
                                break;
                        }
                        juego.siguientePasoCompletado(operacion);
                        break;
                    }
                }
            }
            juego.ranking().toString();
    }
}

File: CivitasRuby/pkg/CivitasRuby-0.0.1/lib/controlador.rb
# encoding:utf-8
require_relative 'civitas_juego'
require_relative 'vista_textual'
require_relative 'operacion_inmobiliaria'

module Civitas
  class Controlador
    def initialize(juego,vista)
      @juego = juego
      @vista = vista
    end

    def juega
      @vista.setCivitasJuego(@juego)
      while !@juego.final_del_juego
        @vista.actualizarVista
        @vista.pausa
        
        operacion = @juego.siguiente_paso
        if operacion != OperacionesJuego::PASAR_TURNO
          @vista.mostrarEventos
        end
        terminado = @juego.final_del_juego
        if !terminado
          case (operacion)
            when OperacionesJuego::COMPRAR
              respuesta = @vista.comprar
              if respuesta == Respuestas::SI
                @juego.comprar
              end
              @juego.siguiente_paso_completado(operacion)
            when OperacionesJuego::GESTIONAR
              @vista.gestionar
              puts "check gestionar"
              op_inmobiliaria = OperacionInmobiliaria.new(@vista.getGestion, @vista.getPropiedad)
              case (@vista.getGestion)
                when GestionesInmobiliarias::LISTA_GESTIONES_INMOBILIARIAS[0]
                  @juego.siguiente_paso_completado(operacion)
                when GestionesInmobiliarias::LISTA_GESTIONES_INMOBILIARIAS[1]
                  @juego.vender(op_inmobiliaria.get_num_propiedad)
                when GestionesInmobiliarias::LISTA_GESTIONES_INMOBILIARIAS[2]
                  @juego.hipotecar(op_inmobiliaria.get_num_propiedad)
                when GestionesInmobiliarias::LISTA_GESTIONES_INMOBILIARIAS[3]
                  @juego.cancelar_hipoteca(op_inmobiliaria.get_num_propiedad)
                when GestionesInmobiliarias::LISTA_GESTIONES_INMOBILIARIAS[4]
                  @juego.construir_casa(op_inmobiliaria.get_num_propiedad)
                when GestionesInmobiliarias::LISTA_GESTIONES_INMOBILIARIAS[5]
                  @juego.construir_hotel(op_inmobiliaria.get_num_propiedad)
              end
              @juego.siguiente_paso_completado(operacion)
            when OperacionesJuego::SALIR_CARCEL
              case(@vista.salir_carcel)
                when SalidasCarcel::LISTA_SALIDAS_CARCEL[0]
                  @juego.salir_carcel_pagando
                when SalidasCarcel::LISTA_SALIDAS_CARCEL[1]
                  @juego.salir_carcel_tirando
              end
              @juego.siguiente_paso_completado(operacion)
          end
        end
      end
      puts @juego.ranking.to_s
    end
  end
end
